book/optimization.qmd

```diff
@@ -1090,34 +1089,111 @@ filter = flt("importance", learner = lrn)
 filter$calculate(task)
 head(as.data.table(filter), 3)
 ```
+### Filter-based Feature Selection
+
+A common usecase for filters is to conduct feature selection based on the filter scores.
+This can be achived using `r ref("mlr_pipeops_filter", text = "PipeOpFilter")`.
+This PipeOp takes as input a Task, applies the filter, and selects the features based on the calculated scores.
+The method how the features are subset can be defined in four different ways, each corresponding to a different parameter:
+
+  * `filter.nfeat` - The *nfeat* values with the highest filter score are selected.
+  * `filter.frac` - The *fraction* of values with the highest filter score are selected.
+  * `filter.cutoff` - All features with a filter score greater than the specified *cutoff* are selected.
+  * `filter.permuted` - A random permutation of each feature is added to the task before applying the filter.
+    All features selected before the *permuted*-th permuted features is selected are kept.
+
+
+:::{.callout-note}
+The `r ref("mlr_pipeops_filter", text = "PipeOpFilter")` only operates on the subset of features that have the correct column type and will leave other columns as is.
+:::
+
+We will illustrate the usage of this filter with the use of a machine learning pipeline for the `r ref("mlr_tasks_optdigits", text = "optdigits")` task, where the goal is to classify handwritten digits.
+We will first subset the features based on the `r ref("mlr3filters::mlr_filters_information_gain", text = "FilterInformationGain")` and then fit a `r ref("mlr3learners::mlr_learners_classif.lda", text = "LDA")`.
+We will tune the parameter `filter.frac` with a simple grid search and visualize the classification error for the different fractions.
+
+```{r optimization-060}
+# reduce logging output
+lgr::get_logger("bbotk")$set_threshold("warn")
+lgr::get_logger("mlr3")$set_threshold("warn")
+
+task = tsk("optdigits")
+
+future::plan(future::multisession(workers = 4))
+
+graph = po("filter", filter = flt("information_gain"), filter.frac = to_tune(0.1, 1)) %>>%
+  po("removeconstants") %>>%
+  po("scale") %>>%
+  po("learner", learner = lrn("classif.lda"))
+
+learner = as_learner(graph)
+
+lda_tuned = auto_tuner(
+  method = tnr("grid_search", resolution = 10),
+  learner = learner,
+  resampling = rsmp("holdout")
+)
+```
+
+:::{.callout-note}
+Unless you have a very specific reason, the parameter controlling how many features are selected, should be tuned.
```

Comment:
Maybe phrase it a bit different: "outside of tutorials, [...]" 

The average user usually doesn't know whether they have a "very specific reason" :)